code/game/objects/structures/signs/_signs.dm

```diff
@@ -61,6 +61,9 @@
 
 /obj/structure/sign/wrench_act(mob/living/user, obj/item/wrench/I)
 	. = ..()
+	//If it's not buildable, just make them hit it with the wrench.
+	if(!buildable_sign)
+		return FALSE
```

Comment:
This is how I was originally going to fix it, but then I realised that there was no way to remove the painting frame without smashing it. That's annoying, so I fixed it on the painting type itself, but I left this in as the comment on this var says that if this var isn't true, you shouldn't be able to unwrench it.